### PR TITLE
Add pylint and unittests

### PR DESCRIPTION
**Pull request by martialblog**
- Add automated unittests from testdata already present in the repo. Includes a small refactor for the CLI to ease testing
- Add pylint and fix minor linting issues
- Change default return code when asking for the version to 3 (UNKNOWN)
- Change validation of HTTP Return code. Now allows for everything below 400
- Change `--mode` arg to be a choice, so that the user knows what can be selected

### CODE REVIEW

**Comment by martialblog** [comment]
TODO: UTC on runners causes tests to fail

**Comment by martialblog** [comment]
Set constant timezone for tests

File: check_vmware_nsxt.py [hunk 1]
```diff
@@ -115,17 +113,17 @@ def request(self, url, method='GET'):
         self.logger.debug("starting API %s request from: %s", method, url)
 
         try:
-            response = requests.request(method, request_url, auth=HTTPBasicAuth(self.username, self.password), verify=self.verify)
+            response = requests.request(method, request_url, auth=HTTPBasicAuth(self.username, self.password), verify=self.verify, timeout=10)
         except requests.exceptions.RequestException as req_exc:
-            raise CriticalException(req_exc)
+            raise CriticalException(req_exc) # pylint: disable=raise-missing-from
 
-        if response.status_code != 200:
```
Comment (RincewindsHat) [inline check_vmware_nsxt.py:122]: What happens if we run into one of the `300` range codes? Is that possible?
`Moved Permanently` and stuff like that.

**Review by RincewindsHat — COMMENTED** [review]
One small comment

File: check_vmware_nsxt.py (hunk 1 above)
Comment (martialblog) [reply check_vmware_nsxt.py:122]: That is why I changed it to ">= 400"

Comment (RincewindsHat) [reply check_vmware_nsxt.py:122]: Exactly, if the response is `301` or something like that, you won't get a proper response either.

Comment (martialblog) [reply check_vmware_nsxt.py:122]: Suggestions? ">= 300"?

Comment (RincewindsHat) [reply check_vmware_nsxt.py:122]: honestly, `!= 200` was probably pretty on point. If you get `200` you should have proper response

Comment (martialblog) [reply check_vmware_nsxt.py:122]: :heavy_check_mark: Done

**Review by RincewindsHat — APPROVED** [review]
Go go go

**Pull request merged**